### DISCUSSION TITLE
Es werden keine Werte im OS angezeigt

### DISCUSSION THREAD

**Discussion by westi2000** [freakent/dbus-mqtt-devices#112] [Q&A]
Hallo, ich habe folgndes Problem, leider bekomme ich keine Werte angezeigt im OS. Ich nutze das Cerbo GX.  Ich bin leider Newbee in Sachen NodeRed und finde den Fehler nicht. Vielleicht ist hier jemand der mir weiterhelfen kann. Bei der Installation sind keine Fehler aufgetreten. 

![grafik](https://github.com/freakent/dbus-mqtt-devices/assets/114467913/aab7419d-fce6-4a4b-89aa-be0e77e11405)

![grafik](https://github.com/freakent/dbus-mqtt-devices/assets/114467913/c735f838-a2f2-4bb7-b956-697c6c5ec9c5)

![grafik](https://github.com/freakent/dbus-mqtt-devices/assets/114467913/c199289e-5554-42c8-86f8-bebc66a59f28)

![grafik](https://github.com/freakent/dbus-mqtt-devices/assets/114467913/83ec3a44-943a-46c7-8bf5-b5dc5427e018)

Bulid Payload

![grafik](https://github.com/freakent/dbus-mqtt-devices/assets/114467913/c43b3fc7-d066-4179-aea2-2cc0e03e9528)

Ac/Power

![grafik](https://github.com/freakent/dbus-mqtt-devices/assets/114467913/cbd9068d-01f0-43d6-b4be-c9cfe8d1317d)

[Gesicherte Terminal-Ausgabe.txt](https://github.com/freakent/dbus-mqtt-devices/files/14550627/Gesicherte.Terminal-Ausgabe.txt)

Installiert habe ich es, wie hier von Matthias beschrieben wurde https://www.youtube.com/watch?v=QTDAZXycTlk&t=1335s

**Comment by freakent** [comment]
Hallo, es schon viele Jahre her ist, dass ich Deutsch studiert habe, werde ich auf Englisch weitermachen.

Hi and thanks for your interest in this project. 

It's not clear from your screenshot where your data source is coming from. It looks like you are subscribed to a topic, reformatting the output then publishing to another topic. But what is publishing to the source topic?

![image](https://github.com/freakent/dbus-mqtt-devices/assets/124850/2b46d3c9-9c20-42c2-ae9b-83a0175b093b)

I suggest adding some debug nodes to your inputs so you can see the when your flow is being vokked an dcheck the payloads.

**Comment by freakent** [comment]
I've just read your code again and looked closer at the screen shots. I see the status messages under the AC/* function nodes. You should look in the NodeRed Context Viewer to see if portalId and ServiceId are being save correctly to the context.
I also think you should have a debug node connected to the output of Build Payload  so you can see if your error message is being output. Can you share also the code in Check Portal ID and Check Service ID?

**Comment by westi2000** [comment]
Hello freakent, thank you very much for your support. I'm currently still at work. I'll test it as soon as I get home. Thanks

**Comment by westi2000** [comment]
Hello, 
here the code  “Check Portal ID” and “Check Service ID”.

<img width="915" alt="grafik" src="https://github.com/freakent/dbus-mqtt-devices/assets/114467913/62db9348-c734-468c-b5d3-06ea5230afe0">
<img width="866" alt="grafik" src="https://github.com/freakent/dbus-mqtt-devices/assets/114467913/acb97cce-d096-4867-9ea0-1d8f50a7c728">
<img width="883" alt="grafik" src="https://github.com/freakent/dbus-mqtt-devices/assets/114467913/f6f94f0c-1902-4b10-8227-7da7acf1c48e">
<img width="876" alt="grafik" src="https://github.com/freakent/dbus-mqtt-devices/assets/114467913/74915bdb-2c2c-4a66-9448-bd811a1fd34a">
<img width="1392" alt="grafik" src="https://github.com/freakent/dbus-mqtt-devices/assets/114467913/7bf552bb-caf8-4122-873a-be6b6f331228">

**Comment by freakent** [comment]
Hi @westi2000,
if you check the example in the README you can see the format of the `device/<client id></DBus` message you should be picking up in your Check and Setze nodes.

```
Topic: "device/fe001/Status"
Payload: { "clientId": "fe001", "connected": 1, "version": "v1.0 ALPHA", "services": {"t1": "temperature"} }
```

```
Topic: "device/<client id>/DBus"
Payload: {"portalId": "<vrm portal id>", deviceInstance":"t1": 5}
```
where t1 is the short name you defined for the service in your Status message.

I have a couple of things to point out:
1) Whilst you can call your variables what you like, it's not a service id you are looking for it's a device instance (deiceInstance is the name Victron use on the dbus).
2) The deviceInstance in the response is a collection so you would need to use `msg.payload.deviceInstance["<short name>"]`
3) It looks like you may have had a typo in your "Setze Servic ID" node, your flow variable name has an "e" on the end. Not sure if that's what you  wanted.
4) I'm not sure what the Check nodes are doing but it looks to me you could combine all 4 check and setze nodes into one node to simplify things a bit.
5) use a seperate debug statement for the step before Check so you can see exactly what's coming out of Check nodes. It looks to me like you are not receiving any data from the resgitration at the moment.

**Comment by westi2000** [comment]
@freakent, thanks for the support, I have to keep looking where my mistake is. As I said, I'm a newbie and have only recently started using NodeRed.

**Reply by freakent** [reply]
Did you see the sample I posted here ?

https://github.com/freakent/dbus-mqtt-devices/discussions/100

It's not perfect but it might help you.
